packages/libs/types/src/Predicate.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * @alpha
+ */
+export const Predicate = {
```

Comment:
why do we need this object?

Reply:
we need this readonly object, because we do not like typescript enums, but theoretically the predicate is a string enum. The values of this enum are used all over the codebase, so it is cleaner to use the enum than the hardcoded string values.

Reply:
I think `TPredicate` should be enough. 
then TS only allows those three valid options and your IDE suggest the types to you

Reply:
That was my initial implementation, but then this further refactor was suggested.